test/integration/apiserver/apiserver_test.go

```diff
@@ -3352,3 +3353,23 @@ func assertManagedFields(t *testing.T, obj *unstructured.Unstructured) {
 func int32Ptr(i int32) *int32 {
 	return &i
 }
+
+func parseFeatureGates(metricsBody string) (map[string]int, error) {
+	featureGates := make(map[string]int)
+	re := regexp.MustCompile(`(?m)^kubernetes_feature_enabled{name="([^"]+)",stage="[^"]*"} (\d+)`)
+	matches := re.FindAllStringSubmatch(metricsBody, -1) // Find all matches
+
+	for _, match := range matches {
+		if len(match) != 3 { // Each match should have the full match and two capture groups
+			return nil, fmt.Errorf("error parsing feature gates where match did not have expected length of 3: %s", match)
+		}
+		featureName := match[1]
+		valueStr := match[2]
+		value, err := strconv.Atoi(valueStr)
+		if err != nil {
```

Comment:
Question still open on how this `err != nil` case is reached: https://github.com/kubernetes/kubernetes/pull/128138#discussion_r1819969614

Reply:
Apologies now, I understand now what this comment was referring to.  I've changed this to return an error now:

```
		if err != nil {
			return nil, fmt.Errorf("unable to convert value %s to integer for feature gate %s", valueStr, featureName)
		}
```